### DISCUSSION TITLE
Method to set class

### DISCUSSION THREAD

**Discussion by ghost** [ueberdosis/tiptap#2162] [General Questions]
I'm trying to give users (limited) options to add certain color to text.
I could do it easily with Tiptap's color extension `@tiptap/extension-color` but, later I thought maybe only injecting a class is more sustainable because the data is being used in other system and class takes up less space when saving (e.g. `class="red"` vs `style="color:rgb(187, 0, 0);"`.)

The part that is stuck is the function I want to put in the `action` part in the code below. I can see some methods in the doc, I can find its source code, tried a few statements but was not very successful. 

```javascript
<template>
 <!--Will iterate stylingBtns to render buttons / use its 'action'-->
</template>

<script>
import { Editor, EditorContent } from '@tiptap/vue-2';

export default {
  name: 'RichEditor',
  components: {
    EditorContent,
  },
  data() {
    return {
     editor: null,
     stylingBtns: [
        {
          button: '<button>Red</button>';
          action: () => // find some method that can toggle a class 'tt-red' on click,
          isActive: () => this.editor.isActive('textStyle', { color: 'rgb(187, 0, 0)' }),
        }, 
    ],
   };
 },
 mounted() {
  // initialize Editor
 },
}
</script>

<style lang="stylus">
.tt-red
  color rgb(187, 0, 0)
</style>

```

**Comment by hanspagel** [comment]
You can add a custom "class" attribute, look for custom attributes in the docs.

I’ve recorded a video to explain the whole process:
https://tiptap.dev/screencasts/building-custom-extensions/build-a-custom-class-extension
